### PR TITLE
Holder Presentation DB models and basic state tracking

### PR DESCRIPTION
**Pull request by usingtechnology**
Signed-off-by: Jason Sherman <tools@usingtechnolo.gy>

Going to go with a small, functional PR. This is the basis for completing Holder functionality in v1, just adding the db model and the protocol listener that will create a record and update the state from AcaPy events.

No API endpoints added, this is the next PR.

### CODE REVIEW

File: services/traction/api/db/models/v1/holder.py [hunk 1]
```diff
@@ -286,3 +286,214 @@ async def get_by_revocation_ids(
...
+        db: AsyncSession,
+        tenant_id: uuid.UUID,
+        thread_id: str,
+    ) -> List["HolderPresentation"]:
+        """List by Thread ID.
+
+        Find and return list of Holder Presentation records for Thread ID.
+
+          tenant_id: Traction ID of tenant making the call
+          thread_id: AcaPy Thread ID of Issuer Presentation
```
Comment (loneil) [inline services/traction/api/db/models/v1/holder.py:485]: Is this supposed to read "Holder Presentation"? Or is Issuer right for this one...

Comment (usingtechnology) [reply services/traction/api/db/models/v1/holder.py:485]: nope, copy/paste...

Comment (usingtechnology) [reply services/traction/api/db/models/v1/holder.py]: fixed.

**Review by loneil — APPROVED** [review]
Looks good

**Pull request merged**